examples/vision/semisupervised_simclr.py

```diff
@@ -0,0 +1,743 @@
+[reported](https://arxiv.org/abs/1911.05722) that the usage of BatchNorm layers could
+sometimes degrade performance, as it introduces an intra-batch dependency between
+samples, which is why I did not have used them in this example. In my experiments
+however, using BatchNorm, especially in the projection head, improves performance.
+
+### Hyperparameters:
+
+The hyperparameters of this example have been tuned manually for this task and
+architecture, so without changing those, only marginal gains can be expected from further
+hyperparameter tuning.
+
+However for a different task or model architecture these would need tuning, so here are
+my notes on the most important ones:
+- **batch size**: since the objective can be interpreted as a classification over a batch
+of images (loosely speaking), batch size is actually a more important hyperparameter than
+usual. The higher, the better.
+- **temperature**: the temperature defines "softness" of the of the softmax distibution
+that is used in the cross-entropy loss, and is an important hyperparameter. Lower values
+generally lead to a higher contrastive accuracy. A recent trick (in
+[ALIGN](https://arxiv.org/pdf/2102.05918.pdf)) is to learn the temperature's value as
+well (which can be done by defining it as a tf.Variable, and applying gradients on it).
+This provides a good baseline value, however in my experiments the learned temperature
+was somewhat lower than optimal, as it is optimized w.r.t. the contrastive loss, which is
+not a perfect proxy for representation quality.
+- **image augmentation strength**: stronger augmentations increase the difficulty of the
+task, however after a point too strong augmentations will degrade performance.
+- **learning rate schedule**: a constant schedule is used here, however it is quite
+common in the literature to use a [cosine decay
+schedule](https://www.tensorflow.org/api_docs/python/tf/keras/experimental/CosineDecay),
+which can further improve performance.
+- **optimizer**: Adam is used in this example, as it provides good performance with
+default parameters. SGD with momentum requires more tuning, however it could slightly
+increase performance.
+"""
+
+"""
+## Related works
+
+Other instance-level (image-level) contrastive learning methods:
+- [MoCo](https://arxiv.org/abs/1911.05722) ([v2](https://arxiv.org/abs/2003.04297),
+[v3](https://arxiv.org/abs/2104.02057)): uses a momentum-encoder as well, whose weights
+are an exponential moving average of the target encoder
+- [SwAV](https://arxiv.org/abs/2006.09882): uses clustering instead of pairwise comparison
+- [BarlowTwins](https://arxiv.org/abs/2103.03230): uses a cross correlation-based
+objective instead of pairwise comparison
+
+I have implemented **MoCo** and **BarlowTwins** as well in a similar fashion, you can
+find these three implementations together in [this
+repository](https://github.com/beresandras/contrastive-classification-keras). A Colab
+Notebook is included as well.
```

Comment:
I would suggested simply saying: Keras implementation  **MoCo** and **BarlowTwins** implementation can be found here. Instead of making a special case for your own workl.